### PR TITLE
S1t9cut js validation

### PR DESCRIPTION
**Pull request by erc-clarkson**
Validation, Sanitisation and API created for Add trainer form.

### CODE REVIEW

File: public/js/addTrainer.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    let formIsValid = true;
+
+    document.querySelectorAll('.formItem_alert').forEach(element => {
+        element.classList.remove('alert-danger');
+        element.classList.add('hidden');
+        element.innerHTML = '';
+    });
+    
+    Object.keys(validate).forEach(formItem => {
+        console.log(formItem)
```
Comment (aidanmaskell) [inline public/js/addTrainer.js:19]: console.log can be removed

File: public/js/addTrainer.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            method: 'post',
+            body: JSON.stringify(data)
+        })
+            .then(response => response.json())
+            .then(responseJson => {
+                if (responseJson.success) {
+                    trainerForm.elements['name'].value = '',
+                    trainerForm.elements['email'].value = '',
+                    trainerForm.elements['notes'].value = '',
+                    message.innerText = responseJson.message,
```
Comment (aidanmaskell) [inline public/js/addTrainer.js:50]: error thrown on this line

File: public/js/addTrainer.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        email: {
+            validLengthVarChar: varCharMaxLength(data.email),
+            isEmail: isEmail(data.email),
+            isPresent: isPresent(data.email)
+        },
+        notes: {
+            validLengthText: textAreaMaxLength(data.notes)
+        }
+    };
+    console.log(validate)
```
Comment (aidanmaskell) [inline public/js/addTrainer.js:91]: console log can be removed

**Review by aidanmaskell — CHANGES_REQUESTED** [review]
General Comments 
- Allows invalid emails through even if front end displays error
- allows duplicate trainers in db

File: public/js/validations.js [hunk 4]
```diff
@@ -33,6 +33,7 @@ function isPostcode(postcode) {
     let regEx = /\b((?:(?:gir)|(?:[a-pr-uwyz])(?:(?:[0-9](?:[a-hjkpstuw]|[0-9])?)|(?:[a-hk-y][0-9](?:[0-9]|[abehmnprv-y])?)))) ?([0-9][abd-hjlnp-uw-z]{2})\b/ig;
     return regEx.test(postcode);
 }
+
```
Comment (mikeycodingstuff) [inline public/js/validations.js:36]: empty line has been added here

**Review by mikeycodingstuff — CHANGES_REQUESTED** [review]
when you successfully submit a trainer and get the success message, the message does not go away if you then submit an invalid trainer and get an error message.

Also,  you can submit a name which is just numbers - not sure if this is an issue

**Pull request merged**